cfn/cfn.go

```diff
@@ -135,37 +139,21 @@ func invoke(handlerFn handlerFunc, request handler.Request, metricsPublisher *me
 		// Create a channel to received a signal that work is done.
 		ch := make(chan handler.ProgressEvent, 1)
 
-		// Create a channel to received error.
-		cherror := make(chan error, 1)
-
 		// Ask the goroutine to do some work for us.
 		go func() {
 			//start the timer
 			start := time.Now()
-			if err := metricsPublisher.PublishInvocationMetric(time.Now(), string(action)); err != nil {
-				cherror <- err
-			}
+			metricsPublisher.PublishInvocationMetric(time.Now(), string(action))
```

Comment:
Log the error rather than just throw it away?

Reply:
I think so. The JAVA and Python plugins log the error.

https://github.com/aws-cloudformation/cloudformation-cli-python-plugin/blob/f195602b2402407ccc5b6210771d14db5d9e25ca/src/cloudformation_cli_python_lib/metrics.py#L51

https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/blob/7460c2a195e1c0fcf487c0c07f6afe234418141a/src/main/java/software/amazon/cloudformation/metrics/MetricsPublisherImpl.java#L125

Reply:
Ok, sounds like we need to do that then :+1: